lib/Dialect/Lattigo/IR/LattigoCKKSOps.td

```diff
@@ -0,0 +1,166 @@
+#ifndef LIB_DIALECT_LATTIGO_IR_LATTIGOCKKSOPS_TD_
+#define LIB_DIALECT_LATTIGO_IR_LATTIGOCKKSOPS_TD_
+
+include "LattigoDialect.td"
+include "LattigoTypes.td"
+include "mlir/IR/OpBase.td"
+
+class Lattigo_CKKSOp<string mnemonic, list<Trait> traits = []> :
+        Lattigo_Op<"ckks." # mnemonic, traits> {
+}
+
+def Lattigo_CKKSNewPlaintextOp : Lattigo_CKKSOp<"new_plaintext"> {
+  let summary = "Create a new plaintext in the Lattigo CKKS dialect";
+  let description = [{
+    This operation creates a new plaintext value in the Lattigo CKKS dialect.
+  }];
+  let arguments = (ins
+    Lattigo_CKKSParameter:$params
+  );
+  let results = (outs Lattigo_RLWEPlaintext:$plaintext);
+}
+
+def Lattigo_CKKSNewParametersFromLiteralOp : Lattigo_CKKSOp<"new_parameters_from_literal"> {
+  let summary = "Create new CKKS parameters from a literal in the Lattigo CKKS dialect";
+  let description = [{
+    This operation creates new CKKS parameters from a given literal value in the Lattigo CKKS dialect.
+  }];
+  let arguments = (ins
+    Lattigo_CKKSParametersLiteral:$paramsLiteral
+  );
+  let results = (outs Lattigo_CKKSParameter:$params);
+}
+
+def Lattigo_CKKSNewEncoderOp : Lattigo_CKKSOp<"new_encoder"> {
+  let summary = "Create a new encoder in the Lattigo CKKS dialect";
+  let description = [{
+    This operation creates a new encoder for encoding plaintext values in the Lattigo CKKS dialect.
+  }];
+  let arguments = (ins
+    Lattigo_CKKSParameter:$params
+  );
+  let results = (outs Lattigo_CKKSEncoder:$encoder);
+}
+
+def Lattigo_CKKSEncodeOp : Lattigo_CKKSOp<"encode"> {
+  let summary = "Encode a plaintext value in the Lattigo CKKS dialect";
+  let description = [{
+    This operation encodes a plaintext value using the specified encoder in the Lattigo CKKS dialect.
+  }];
+  let arguments = (ins
+    Lattigo_CKKSEncoder:$encoder,
+    AnyType:$value,
```

Comment:
why not use RankedTensorType here instead of using the custom verifier? (Here and for the decodeop result)

Reply:
Changed the AnyType to AnyRankedTensor. Verifier still needed for element type though.

Reply:
You should be able to simplify this further by using `RankedTensorOf<[AnyInteger]>`, and for the float/complex cases you can give `RankedTensorOf<[AnyFloat, Complex<AnyFloat>]>` I think this is enough to remove the verifiers in many of these ops.

Reply:
Used `RankedTensorOf<[AnyInteger]>` and `AllTypesMatch` to remove the verifiers.